scraper/maiia/maiia.py

```diff
@@ -1,117 +1,120 @@
-    now = datetime.now()
-    start_date = datetime.strftime(now, '%Y-%m-%dT%H:%M:%S.%f%zZ')
-    end_date = (now + timedelta(days=MAIIA_DAY_LIMIT)).strftime('%Y-%m-%dT%H:%M:%S.%f%zZ')
 
-    url = f'https://www.maiia.com/api/pat-public/availabilities?centerId={center_id}&from={start_date}&to={end_date}&page=0&limit={MAIIA_LIMIT}'
-    req = session.get(url)
-    req.raise_for_status()
-    data = req.json()
-    items = data.get('items', [])
-    slots = []
-    for item in items:
-        timeslot = item.get('timeSlotId', None)
-        if not timeslot:
-            continue
-        if timeslot not in slots:
-            slots.append(timeslot)
-    return len(slots)
+def get_next_slot_date(center_id: str, consultation_reason_name: str, start_date: str, client: httpx.Client = DEFAULT_CLIENT) -> str:
+    url = f'{MAIIA_URL}/api/pat-public/availability-closests?centerId={center_id}&consultationReasonName={consultation_reason_name}&from={start_date}'
+    try:
+        r = client.get(url)
+        r.raise_for_status()
+    except httpx.HTTPStatusError as hex:
+        logger.warning(f'{url} returned error {hex.response.status_code}')
+        return
+    result = r.json()
+    if 'firstPhysicalStartDateTime' in result:
+        return result['firstPhysicalStartDateTime']
+    return
+
+
+def get_slots(center_id: str, consultation_reason_name: str, start_date: str, end_date: str, limit=MAIIA_LIMIT, client: httpx.Client = DEFAULT_CLIENT) -> list:
+    url = f'{MAIIA_URL}/api/pat-public/availabilities?centerId={center_id}&consultationReasonName={consultation_reason_name}&from={start_date}&to={end_date}'
+    availabilities = get_paged(url, limit=limit, client=client)['items']
+    if not availabilities:
+        next_slot_date = get_next_slot_date(
+            center_id, consultation_reason_name, start_date, client=client)
+        if not next_slot_date:
+            return
+        next_date = datetime.strptime(next_slot_date, '%Y-%m-%dT%H:%M:%S.%fZ')
+        start_date = datetime.strftime(next_date, '%Y-%m-%dT%H:%M:%S.%f%zZ')
+        end_date = (next_date + timedelta(days=MAIIA_DAY_LIMIT)
+                    ).strftime('%Y-%m-%dT%H:%M:%S.%f%zZ')
+        url = f'{MAIIA_URL}/api/pat-public/availabilities?centerId={center_id}&consultationReasonName={consultation_reason_name}&from={start_date}&to={end_date}'
+        availabilities = get_paged(url, limit=100000, client=client)['items']
+    if availabilities:
+        return availabilities
+    return
+
+
+def get_reasons(center_id: str, limit=MAIIA_LIMIT, client: httpx.Client = DEFAULT_CLIENT) -> list:
+    url = f'{MAIIA_URL}/api/pat-public/consultation-reason-hcd?rootCenterId={center_id}'
+    result = get_paged(url, limit=limit, client=client)
+    if not result['total']:
+        return []
+    return result['items']
+
+
+def get_first_availability(center_id: str, request_date: str, reasons: dict, client: httpx.Client = DEFAULT_CLIENT) -> [datetime, int]:
```

Comment:
A fortiori, `reasons` vient de `get_reasons()` qui renvoie une `list de `dict`.
```suggestion
def get_first_availability(center_id: str, request_date: str, reasons: list[dict], client: httpx.Client = DEFAULT_CLIENT) -> [datetime, int]:
```

Reply:
Apparemment on peut retourner `None, 0`, donc même : 
```suggestion
def get_first_availability(center_id: str, request_date: str, reasons: list[dict], client: httpx.Client = DEFAULT_CLIENT) -> [Optional[datetime], int]:
```